pkg/placement/apiwatch-map-provider.go

```diff
@@ -92,10 +88,25 @@ func (awp *apiWatchProvider) AddReceivers(clusterName logicalcluster.Name,
 			groupReceivers:    MappingReceiverHolderFork[string /*group name*/, APIGroupInfo]{},
 			resourceReceivers: MappingReceiverHolderFork[metav1.GroupResource, ResourceDetails]{},
 		}
-		discoveryScopedClient := awp.discoveryClusterClient.Cluster(clusterName.Path())
-		crdInformer := awp.crdClusterPreInformer.Cluster(clusterName).Informer()
-		bindingInformer := awp.bindingClusterPreInformer.Cluster(clusterName).Informer()
-		wpc.informer, wpc.lister, _ = apiwatch.NewAPIResourceInformer(ctx, clusterName.String(), discoveryScopedClient, false, crdInformer, bindingInformer)
+
+		config, err := awp.spaceclient.ConfigForSpace(clusterName, awp.spaceProviderNs)
+		if err != nil {
+			logger.Error(err, "Failed to get space config", "space", clusterName)
+		}
+		discoveryScopedClient, err := discovery.NewDiscoveryClientForConfig(config)
+		if err != nil {
+			logger.Error(err, "Failed to create discovery client", "space", clusterName)
+		}
+
+		apiextClient, err := apiextclient.NewForConfig(config)
+		if err != nil {
+			logger.Error(err, "Failed to create clientset for CustomResourceDefinitions")
+		}
+		apiextFactory := apiextinfactory.NewSharedInformerFactory(apiextClient, 0)
```

Comment:
Since this factory is not used for anything else, it would be simpler (and less wasteful) to directly call `NewCustomResourceDefinitionInformer`.